// @profabric/react-components/dist/index.js
import React from "react";
import { createComponent } from "./lib/create-components";
import * as Components from "@profabric/web-components";
export const Badge = createComponent({
    react: React,
    displayName: "Badge",
    tagName: "pf-badge",
    elementClass: Components.Badge,
    events: {},
});
export const Button = createComponent({
    react: React,
    displayName: "Button",
    tagName: "pf-button",
    elementClass: Components.Button,
    events: {
        onClick: 'click'
    },
});
export const Checkbox = createComponent({
    react: React,
    displayName: "Checkbox",
    tagName: "pf-checkbox",
    elementClass: Components.Checkbox,
    events: {
        onChange: 'change'
    },
});
export const Dropdown = createComponent({
    react: React,
    displayName: "Dropdown",
    tagName: "pf-dropdown",
    elementClass: Components.Dropdown,
    events: {},
});
export const Form = createComponent({
    react: React,
    displayName: "Form",
    tagName: "pf-form",
    elementClass: Components.Form,
    events: {},
});
export const Image = createComponent({
    react: React,
    displayName: "Image",
    tagName: "pf-image",
    elementClass: Components.Image,
    events: {},
});
export const Input = createComponent({
    react: React,
    displayName: "Input",
    tagName: "pf-input",
    elementClass: Components.Input,
    events: {
        onChange: 'change'
    },
});
export const Select = createComponent({
    react: React,
    displayName: "Select",
    tagName: "pf-select",
    elementClass: Components.Select,
    events: {
        onChange: 'change'
    },
});
export const Spinner = createComponent({
    react: React,
    displayName: "Spinner",
    tagName: "pf-spinner",
    elementClass: Components.Spinner,
    events: {},
});
